plugins/statistics.koplugin/main.lua

```diff
@@ -2661,12 +2713,24 @@ end
 
 
 function ReaderStatistics:onPosUpdate(pos, pageno)
+    if self.use_pagemap_for_stats then
+        local current_page = select(2, self.ui.pagemap:getCurrentPageLabel())
+        if self.curr_page ~= current_page then
+            self:onPageUpdate(pageno)
+        end
+        return
+    end
+
     if self.curr_page ~= pageno then
         self:onPageUpdate(pageno)
     end
 end
 
 function ReaderStatistics:onPageUpdate(pageno)
+    if self.use_pagemap_for_stats and pageno ~= false then
+        pageno = select(2, self.ui.pagemap:getCurrentPageLabel())
+    end
+
     if not self:isEnabledAndNotFrozen() then
         return
     end
```

Comment:
I think you can put your 4 added lines after the isEnabledAndNotFrozen() (if we return, no need fetch pagno).